src/vs/server/node/server.cli.ts

```diff
@@ -460,5 +460,6 @@ function mapFileToRemoteUri(uri: string): string {
 }
 
 const [, , productName, version, commit, executableName, ...remainingArgs] = process.argv;
-main({ productName, version, commit, executableName }, remainingArgs);
-
+main({ productName, version, commit, executableName }, remainingArgs).then(null, err => {
```

Comment:
@alexdima @aeschli fyi this makes the `main` function of the server `async`, similar to how `cli.ts` is `async` for desktop.